File: code/bindings/csharp/Tests/ZpeInkParity.cs
using System;
using System.IO;
using ZPE.Ink;

public static class ZpeInkParity {
    public static int Main(string[] args) {
        if (args.Length != 2) {
            Console.Error.WriteLine("usage: mono ZpeInkParity.exe <input.zpink> <expected.json>");
            return 2;
        }

        try {
            var input = File.ReadAllBytes(args[0]);
            var expected = File.ReadAllText(args[1]);
            var actual = ZpeInk.DecodeToCanonicalJson(input);
            if (!String.Equals(actual, expected, StringComparison.Ordinal)) {
                Console.Error.WriteLine("csharp parity mismatch");
                return 1;
            }

            Console.Write(actual);
            return 0;
        } catch (Exception ex) {
            Console.Error.WriteLine(ex.Message);
            return 1;
        }
    }
}

File: code/bindings/swift/Tests/ZPEInkParity.swift
import Foundation

@main
struct ZPEInkParity {
    static func main() throws {
        let arguments = CommandLine.arguments
        guard arguments.count == 3 else {
            fputs("usage: swift ZPEInkParity.swift <input.zpink> <expected.json>\n", stderr)
            Foundation.exit(2)
        }

        let inputURL = URL(fileURLWithPath: arguments[1])
        let expectedURL = URL(fileURLWithPath: arguments[2])
        let payload = try Data(contentsOf: inputURL)
        let expected = try String(contentsOf: expectedURL, encoding: .utf8)
        let actual = try ZPEInk.decodeToCanonicalJSON(payload)

        guard actual == expected else {
            fputs("swift parity mismatch\n", stderr)
            Foundation.exit(1)
        }

        print(actual, terminator: "")
    }
}
